python/lsst/ts/ATHexapod/csc.py

```diff
@@ -0,0 +1,165 @@
+        self.evt_detailedState.set_put(detailedState=self.detailed_state)
+
+    def get_config_pkg(self):
+        """Return the name of the config directory for the ATHexapod CSC."""
+        return "ts_config_attcs"
+
+    async def configure(self, config):
+        """Configure the CSC.
+
+        Parameters
+        ----------
+        config
+        """
+        pass
+
+    async def end_enable(self, id_data):
+        """Set the detailed_state as NOTINMOTION.
+
+        Parameters
+        ----------
+        id_data
+        """
+        self.detailed_state = ATHexapod.DetailedState.NOTINMOTION
+
+    async def do_applyPositionLimits(self, id_data):
+        """Apply the position limits.
+
+        Parameters
+        ----------
+        id_data
+        """
+        self.assert_enabled("applyPositionLimits")
+        self.assert_substate([ATHexapod.DetailedState.NOTINMOTION], "applyPositionLimits")
+        # add model call
+
+    async def do_moveToPosition(self, id_data):
+        """Move the Hexapod to position.
+
+        Parameters
+        ----------
+        id_data
+        """
+        self.assert_enabled("moveToPosition")
+        self.assert_substate([ATHexapod.DetailedState.NOTINMOTION], "moveToPosition")
+        self.detailed_state = ATHexapod.DetailedState.INMOTION
+        # add model call
+        self.detailed_state = ATHexapod.DetailedState.NOTINMOTION
+
+    async def do_setMaxSystemSpeeds(self, id_data):
+        """Set max system speeds.
+
+        Parameters
+        ----------
+        id_data
+        """
+        self.assert_enabled("setMaxSystemSpeeds")
+        self.assert_substate([ATHexapod.DetailedState.NOTINMOTION], "setMaxSystemSpeeds")
+        # add model call
+
+    async def do_applyPositionOffset(self, id_data):
+        """Apply position offset.
+
+        Parameters
+        ----------
+        id_data
+        """
+        self.assert_enabled("applyPositionOffset")
+        self.assert_substate([ATHexapod.DetailedState.NOTINMOTION], "applyPositionOffset")
+        # add model call
+
+    async def do_pivot(self, id_data):
+        """Pivot the hexapod.
+
+        Parameters
+        ----------
+        id_data
+        """
+        self.assert_enabled("pivot")
+        self.assert_substate([ATHexapod.DetailedState.NOTINMOTION], "pivot")
+        self.detailed_state = ATHexapod.DetailedState.INMOTION
+        # add model call
+        self.detailed_state = ATHexapod.DetailedState.NOTINMOTION
+
+    async def do_stopAllAxes(self, id_data):
+        """Stop all axes.
+
+        Parameters
+        ----------
+        id_data
+        """
+        self.assert_enabled("stopAllAxes")
+        self.assert_substate([ATHexapod.DetailedState.INMOTION], "stopAllAxes")
+        # add model call
+        self.detailed_state = ATHexapod.DetailedState.NOTINMOTION
+
+    async def telemetry(self):
+        """Handle telemetry publishing."""
+        pass
+
+    @classmethod
+    async def amain(cls):
```

Comment:
I don't think you need to override the `amain` method here and it doesn't seem like you are doing anything special with it.